CRM/MembershipExtras/Form/RecurringContribution/AddDonationLineItem.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+use CRM_MembershipExtras_ExtensionUtil as E;
+
+/**
+ * Form controller class
+ *
+ * @see https://wiki.civicrm.org/confluence/display/CRMDOC/QuickForm+Reference
+ */
+class CRM_MembershipExtras_Form_RecurringContribution_AddDonationLineItem extends CRM_MembershipExtras_Form_RecurringContribution_AddLineItem {
+
+  /**
+   * @inheritdoc
+   */
+  public function preProcess() {
+    parent::preProcess();
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function buildQuickForm() {
+    CRM_Utils_System::setTitle(E::ts('Add %1', ['1' => $this->lineItemParams['item']]) . '?');
+
+    parent::buildQuickForm();
+  }
+
+  /**
+   * @inheritdoc
+   */
+  protected function processLineItemAddition() {
+    $recurringLineItem = $this->createRecurringLineItem();
+    $this->addLineItemToPendingContributions($recurringLineItem);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  protected function showOnSuccessNotifications() {
+    CRM_Core_Session::setStatus(
+      "{$this->lineItemParams['item']} has been added to the active order.",
+      "Add {$this->lineItemParams['item']}",
+      'success'
+    );
+
+    if ($this->lineItemParams['auto_renew']) {
+      CRM_Core_Session::setStatus(
+        "{$this->lineItemParams['item']} will now be continued in the next period.",
+        "Add {$this->lineItemParams['item']}",
+        'success'
+      );
+    }
+  }
+
+  /**
+   * @inheritdoc
+   */
+  protected function showErrorNotification(Exception $e) {
+    CRM_Core_Session::setStatus(
+      "An error ocurred trying to add {$this->lineItemParams['item']} to the current recurring contribution: " . $e->getMessage(),
+      "Error Adding {$this->lineItemParams['item']}",
+      'error'
+    );
+  }
+
+  /**
+   * Creates new line item associated to the recurring contribution.
+   *
+   * @return array
+   */
+  protected function createRecurringLineItem() {
+    $taxRate = $this->getTaxRateForFinancialType($this->lineItemParams['financial_type_id']);
+    $taxAmount = round($this->lineItemParams['amount'] * $taxRate / 100, 2);
```

Comment:
can you check all the places that use round($amount/100, 2) formula and move it to a new class (that could probably called AmountCalculator or whatever) instead of repeating the same line in the extension codebase multiple times.